aws-xray-agent-plugin/src/test/java/com/amazonaws/xray/agent/runtime/handlers/downstream/source/sql/MyConnectionImpl.java

```diff
@@ -0,0 +1,302 @@
+package com.amazonaws.xray.agent.runtime.handlers.downstream.source.sql;
+
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.SQLClientInfoException;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+public class MyConnectionImpl implements Connection {
```

Comment:
Confirming you couldn't use a mock for this?

Reply:
That's right, due to a disco limitation where you can't mock and instrument the same method.